app/controllers/api/v1/moves_actions.rb

```diff
@@ -83,7 +84,6 @@ def new_move_attributes
           to_location: Location.find_by(id: new_move_params.dig(:relationships, :to_location, :data, :id)),
           court_hearings: CourtHearing.where(id: (new_move_params.dig(:relationships, :court_hearings, :data) || []).map { |court_hearing| court_hearing[:id] }),
           prison_transfer_reason: PrisonTransferReason.find_by(id: new_move_params.dig(:relationships, :prison_transfer_reason, :data, :id)),
-          supplier: SupplierChooser.new(doorkeeper_application_owner, from_location).call,
```

Comment:
are there any vantages removing this?

this could be 
```
  supplier: determine_supplier
```
with some changes on the `determine_supplier`, that should just return the supplier, no assignment in it.
If this is possible, we could keep some controller's actions cleaner

Reply:
see previous comment. This is encapsulated in the move instance method `determine_supplier` - and needs to be passed either the move or move_date - better to keep that logic within the move model. So that means we need to create the move first, and then call `determine_supplier` on the move.